mediapipe/graphs/instantmotiontracking/calculators/gl_animation_overlay_calculator.cc

```diff
@@ -342,32 +344,45 @@ bool GlAnimationOverlayCalculator::LoadAnimationAndroid(
       float By = V3y - V1y;
       float Bz = V3z - V1z;
       // Calculate cross product
-      triangle_normals[idx] = Ay * Bz - Az * By;
-      triangle_normals[idx + 1] = Az * Bx - Ax * Bz;
-      triangle_normals[idx + 2] = Ax * By - Ay * Bx;
-    }
-
-    // Normalize each triangle surface normal
-    for(int i = 0; i < len; i+=3) {
-      float product = 0.0;
-      product = product + triangle_normals[i] * triangle_normals[i];
-      product = product + triangle_normals[i+1] * triangle_normals[i+1];
-      product = product + triangle_normals[i+2] * triangle_normals[i+2];
-      float magnitude = sqrt(product);
-      triangle_normals[i] /= magnitude;
-      triangle_normals[i+1] /= magnitude;
-      triangle_normals[i+2] /= magnitude;
+      float normal_x = Ay * Bz - Az * By;
+      float normal_y = Az * Bx - Ax * Bz;
+      float normal_z = Ax * By - Ay * Bx;
+      // Add connected normal to each associated vertex
+      // It is also necessary to increment each vertex denominator for averaging
```

Comment:
I think you need to normalize here first too.  Otherwise, when you average a vector of (1, 0, 0) and (0, 0, 1000), you'll get something really close to (0, 0, 1) instead of (1/rt(2), 0, 1/rt(2)).  So let's also pull normalize into a quick separate function so we can reuse easily.